### ISSUE TITLE
Wii U build not working from 10/04 onwards

### ISSUE THREAD

**Issue by Azy77** [libretro/libretro-uae#283]
Hi, I am using nightly from 8th April which works fine but from 10/04 onwards the kickstarts do not load on the Wii U, there is a red screen which changes to black but kickstarts do not load, tested a500, a600, a1200.

**Comment by sonninnos** [comment]
Thanks for reporting.

Is it possible to get stdout log with WiiU? UAE outputs all relevant information there, so unfortunately RA's log will not be of any use.

**Comment by Azy77** [comment]
Not sure what stdout is, how do I do that?

**Comment by sonninnos** [comment]
Sorry, but I have no idea how WiiU works. Stdout/stderr is the stuff that is outputted to the console when running the executable with `--verbose`.

Around that time there was an update that fixed linux-ARM builds, so maybe it has something to do with this. Very hard to say, but if older versions work, then most likely yes.

**Comment by rsn8887** [comment]
to get logs that are saved to retroarch/logs, go to settings->logging. Then enable logs, set log levels to 0 (DEBUG), enable log to file and enable timestamps. Then start the core. After the crash, you should have log files in retroarch/logs/. Read these log files using a PC or laptop and copy/paste them onto pastebin.net. Share the link to pastebin.net here.

**Comment by Azy77** [comment]
Thanks for the info.
The core does not actually crash, it just fails to load the kickstart then when I insert lha then restart there is a red screen then black and it remains black but I can still access the RA menu. 

Would a log still be created?

**Comment by sonninnos** [comment]
Yes, that is not a crash dump log, and it will help if it indeed contains also stdout rows.

Can you navigate to RA saves directory and locate the generated config file `puae_libretro.uae` and post the contents?

**Comment by Azy77** [comment]
I currently have p-uae core from 8th april, I will download latest core then load it up then post the contents of that file.

**Comment by sonninnos** [comment]
Hey @phcoder, might you have any ideas with this, since it appears this started right after the endianness-fixes?

**Comment by crystalct** [comment]
Hi guys, few days ago i started to fix PS3 build of libretro-uae and after accomplished fixing, i tried to start core without succes. After a lot a study i wasn't able to start anything and a doubt started on my mind.... Big endian problems? Let's see in opened issues... bingo! So i take the versione of 8 April and applied same fix of the current version. Bingo again!! Libretro.uae works correctly on PS3. 
Now i will apply next commits after 8 April to understand where and when was introduced a big endian issure (i hope the only one big endian issue).
My work to fix PS3 build on current version is: https://github.com/crystalct/libretro-uae

![image](https://user-images.githubusercontent.com/1675181/96451794-694b7500-1218-11eb-83f7-5f9e718e6aac.png)

![image](https://user-images.githubusercontent.com/1675181/96452091-d19a5680-1218-11eb-8063-ec5d40d9fc50.png)

**Comment by sonninnos** [comment]
Excellent news! Looks like just some cleanup needed before PR readiness.

**Comment by crystalct** [comment]
Found commit that introduced that big endian issue:
https://github.com/libretro/libretro-uae/commit/9c512a45d711e8b5b5126e0a2dd3b48d2806dc95

**Comment by crystalct** [comment]
A not elegant fix....

retrodep/machdep/m68kops.h:
```
#if defined (WIIU) || defined (__CELLOS_LV2__)
#include <retro_endianness.h>
#if RETRO_IS_BIGENDIAN
#define WORDS_BIGENDIAN
#endif
#include "../../sources/src/md-generic/m68kops.h"
#elif defined (__x86_64__)
#include "../../sources/src/md-amd64-gcc/m68kops.h"
#elif defined (__i386__)
#include "../../sources/src/md-i386-gcc/m68kops.h"
#elif defined(__powerpc__)
#include "../../sources/src/md-ppc-gcc/m68kops.h"
#else
#include <retro_endianness.h>
#if RETRO_IS_BIGENDIAN
#define WORDS_BIGENDIAN
#endif
#include "../../sources/src/md-generic/m68kops.h"
#endif
```

retrodep/machdep/m68k.h:
```
#if defined (WIIU) || defined (__CELLOS_LV2__)
#include "../../sources/src/md-generic/m68k.h"
#elif defined (__x86_64__)
#include "../../sources/src/md-amd64-gcc/m68k.h"
#elif defined (__i386__)
#include "../../sources/src/md-i386-gcc/m68k.h"
#elif defined(__powerpc__)
#include "../../sources/src/md-ppc-gcc/m68k.h"
#else
#include "../../sources/src/md-generic/m68k.h"
#endif
```

...becose  `WIIU` and `__CELLOS_LV2__` must be before others or __powerpc__is recognized by PS3 (and i suppose by WIIU too).

**Comment by sonninnos** [comment]
How about instead:
`#elif defined(__powerpc__) && !defined(WIIU) && !defined(__CELLOS_LV2__)`

**Comment by crystalct** [comment]
Yes Is Better. I thought the same thing After my post 🤭

Il lun 19 ott 2020, 17:18 Tony <notifications@github.com> ha scritto:

**Comment by sonninnos** [comment]
@Azy77 Could you test the current WiiU build once it is available?

**Comment by crystalct** [comment]
@vaguerant could you test this core on WiiU?
There wasn't reply from @Azy77 :-(

**Comment by vaguerant** [comment]
Seems to be working on my end, I think. I have no experience emulating Amiga computers, but I was able to boot an A500 game successfully. I tried another game (A600, I think?) but just got booted back to the AmigaDOS screen. Possibly I needed to use different settings, I really don't know what I'm doing. But the core certainly seems to be reading the Kickstart ROMs successfully, so the issue as stated seems resolved to me.
